liblocation/src/main/java/com/mapbox/android/core/location/LocationEngineProxy.java

```diff
@@ -0,0 +1,79 @@
+package com.mapbox.android.core.location;
+
+import android.app.PendingIntent;
+import android.os.Looper;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.annotation.VisibleForTesting;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static com.mapbox.android.core.location.Utils.checkNotNull;
+
+class LocationEngineProxy<T> implements LocationEngine {
+  private final LocationEngineImpl<T> locationEngineImpl;
+  private Map<LocationEngineCallback<LocationEngineResult>, T> listeners;
+
+  LocationEngineProxy(LocationEngineImpl<T> locationEngineImpl) {
+    this.locationEngineImpl = locationEngineImpl;
+  }
+
+  @Override
+  public void getLastLocation(@NonNull LocationEngineCallback<LocationEngineResult> callback) throws SecurityException {
+    checkNotNull(callback, "callback == null");
+    locationEngineImpl.getLastLocation(callback);
+  }
+
+  @Override
+  public void requestLocationUpdates(@NonNull LocationEngineRequest request,
+                                     @NonNull LocationEngineCallback<LocationEngineResult> callback,
+                                     @Nullable Looper looper) throws SecurityException {
+    checkNotNull(request, "request == null");
+    checkNotNull(callback, "callback == null");
+    locationEngineImpl.requestLocationUpdates(request, getListener(callback),
+      looper == null ? Looper.getMainLooper() : looper);
+  }
+
+  @Override
+  public void requestLocationUpdates(@NonNull LocationEngineRequest request,
+                                     PendingIntent pendingIntent) throws SecurityException {
+    checkNotNull(request, "request == null");
+    locationEngineImpl.requestLocationUpdates(request, pendingIntent);
+  }
+
+  @Override
+  public void removeLocationUpdates(@NonNull LocationEngineCallback<LocationEngineResult> callback) {
+    checkNotNull(callback, "callback == null");
+    locationEngineImpl.removeLocationUpdates(removeListener(callback));
+  }
+
+  @Override
+  public void removeLocationUpdates(PendingIntent pendingIntent) {
+    locationEngineImpl.removeLocationUpdates(pendingIntent);
+  }
+
+  @VisibleForTesting
```

Comment:
Method not used for testing. We should remove tag or unnecessary method.

Reply:
it is used by this test class: https://github.com/mapbox/mapbox-events-android/blob/1cfd4c49060f163917333fe1a686fdeb2f43fa0d/liblocation/src/test/java/com/mapbox/android/core/location/LocationEngineProxyTest.java#L15